### PR TITLE
Backend/fix use not filtered location updates

### PR DESCRIPTION
**Pull request by roman-bodavskiy**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Use not filtered waypoints:

- upsert driver location to db/redis
- Trigger` the kafka stream

Use filtered with accuracy waypoints

- Add driver speed in Redis
- Add intermediate waypoints in Redis

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs [hunk 1]
```diff
@@ -143,30 +143,34 @@ updateLocationHandler UpdateLocationHandle {..} waypoints = withLogTag "driverLo
...
-                  updateDriverSpeedInRedis driver.merchantId driver.id point.pt point.ts
-                  streamLocationUpdates (fst <$> mbRideIdAndStatus) driver.merchantId driver.id point.pt point.ts driverInfo.active driverInfo.mode
-              )
-              (a : ax)
+            forM_ (a : ax) $ \point -> do
+              streamLocationUpdates (fst <$> mbRideIdAndStatus) driver.merchantId driver.id point.pt point.ts driverInfo.active driverInfo.mode
+
+      let filteredWaypointWithAccuracy = filter (\val -> fromMaybe 0.0 val.acc <= minLocationAccuracy) filteredWaypoint
+      case filteredWaypointWithAccuracy of
+        [] -> logWarning "Incoming points are older than current one, or points have bad accuracy, ignoring"
```
Comment (sidsethujuspay) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs:162]: This warning should say, "accuracy of the points is low, ignoring".

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs [hunk 2]
```diff
@@ -143,30 +143,34 @@ updateLocationHandler UpdateLocationHandle {..} waypoints = withLogTag "driverLo
...
     LocUpd.whenWithLocationUpdatesLock driver.id $ do
       mbOldLoc <- findDriverLocation
-      case filterNewWaypoints mbOldLoc minLocationAccuracy of
-        [] -> logWarning "Incoming points are older than current one, ignoring"
+      let sortedWaypoint = toList $ NE.sortWith (.ts) waypoints
+          filteredWaypoint = maybe sortedWaypoint (\oldLoc -> filter ((oldLoc.coordinatesCalculatedAt <) . (.ts)) sortedWaypoint) mbOldLoc
+      mbRideIdAndStatus <- getAssignedRide
+
+      case filteredWaypoint of
+        [] -> pure ()
```
Comment (sidsethujuspay) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs:151]: Please add a logwarning that says "Incoming points are older than current one, or points have bad accuracy, ignoring".

Comment (roman-bodavskiy) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs:151]: why? here we didn't check accuracy yet

Comment (sidsethujuspay) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs:151]: Updated the text.

**Review by hkmangla — APPROVED** [review]

**Pull request merged**